sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/SyntaxTrackers/VisualBasicObjectCreationTracker.cs

```diff
@@ -47,5 +48,19 @@ public VisualBasicObjectCreationTracker(IAnalyzerConfiguration analyzerConfigura
                     argumentList.Arguments.Count > index &&
                     argumentList.Arguments[index].GetExpression().IsConstant(context.SemanticModel);
             };
+
+        internal override object ConstArgumentForParameter(ObjectCreationContext context, string parameterName)
+        {
+            var argumentList = ((ObjectCreationExpressionSyntax)context.Expression).ArgumentList;
+            var methodParameterLookup = new VisualBasicMethodParameterLookup(argumentList, context.SemanticModel);
+            var argumentParameterMappingForParameter = methodParameterLookup.GetAllArgumentParameterMappings()
```

Comment:
There's a single method for this in my new rule S4830. It would reduce it to if-statement only.
But I'm not sure what's the good solution for this now. Leave it as is or wait, merge S4830, rebase and update?

Reply:
Leave it as it is and update it in hardening, may be?

Reply:
The methods are a bit different, the one in rule S4830 PR is comparing symbols, whereas here it starts with a string (parameter name). However, the logic is very similar, so I will rebase on master once PR S4830 is merged and refactor this part